File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/EncounterEntity.java
package org.openmrs.mobile.databases.entities;

import org.openmrs.mobile.models.Resource;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;

@Entity(tableName = "encounters")
public class EncounterEntity extends Resource {

    @ColumnInfo(name = "visit_id")
    private String visitKeyId;

    @NonNull
    @ColumnInfo(name = "encounterDatetime")
    private String encounterDateTime;

    @ColumnInfo(name = "type")
    private String encounterType;

    @ColumnInfo(name = "patient_uuid")
    private String patientUuid;

    @ColumnInfo(name = "form_uuid")
    private String formUuid;

    public EncounterEntity() {
    }

    public void setVisitKeyId(String visitKeyId) {
        this.visitKeyId = visitKeyId;
    }

    public void setEncounterDateTime(@NonNull String encounterDateTime) {
        this.encounterDateTime = encounterDateTime;
    }

    public void setEncounterType(String encounterType) {
        this.encounterType = encounterType;
    }

    public void setPatientUuid(String patientUuid) {
        this.patientUuid = patientUuid;
    }

    public void setFormUuid(String formUuid) {
        this.formUuid = formUuid;
    }

    public String getVisitKeyId() {
        return visitKeyId;
    }

    @NonNull
    public String getEncounterDateTime() {
        return encounterDateTime;
    }

    public String getEncounterType() {
        return encounterType;
    }

    public String getPatientUuid() {
        return patientUuid;
    }

    public String getFormUuid() {
        return formUuid;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof EncounterEntity)) return false;

        EncounterEntity that = (EncounterEntity) o;

        if (visitKeyId != null ? !visitKeyId.equals(that.visitKeyId) : that.visitKeyId != null)
            return false;
        if (!encounterDateTime.equals(that.encounterDateTime)) return false;
        if (encounterType != null ? !encounterType.equals(that.encounterType) : that.encounterType != null)
            return false;
        if (patientUuid != null ? !patientUuid.equals(that.patientUuid) : that.patientUuid != null)
            return false;
        return formUuid != null ? formUuid.equals(that.formUuid) : that.formUuid == null;
    }

    @Override
    public int hashCode() {
        int result = visitKeyId != null ? visitKeyId.hashCode() : 0;
        result = 31 * result + encounterDateTime.hashCode();
        result = 31 * result + (encounterType != null ? encounterType.hashCode() : 0);
        result = 31 * result + (patientUuid != null ? patientUuid.hashCode() : 0);
        result = 31 * result + (formUuid != null ? formUuid.hashCode() : 0);
        return result;
    }
}

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/EncounterEntity.kt
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package org.openmrs.mobile.databases.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import org.openmrs.mobile.models.Resource

@Entity(tableName = "encounters")
class EncounterEntity : Resource() {
    @ColumnInfo(name = "visit_id")
    @SerializedName("visit_id")
    @Expose
    var visitKeyId: String? = null

    @ColumnInfo(name = "encounterDatetime")
    @SerializedName("encounterDatetime")
    @Expose
    lateinit var encounterDateTime: String

    @ColumnInfo(name = "type")
    @SerializedName("type")
    @Expose
    var encounterType: String? = null

    @ColumnInfo(name = "patient_uuid")
    @SerializedName("patient_uuid")
    @Expose
    var patientUuid: String? = null

    @ColumnInfo(name = "form_uuid")
    @SerializedName("form_uuid")
    @Expose
    var formUuid: String? = null

    @ColumnInfo(name = "location_uuid")
    @SerializedName("location_uuid")
    @Expose
    var locationUuid: String? = null

    @ColumnInfo(name = "encounter_provider_uuid")
    @SerializedName("encounter_provider_uuid")
    @Expose
    var encounterProviderUuid: String? = null

}